src/WebUI/Common/KubeSummary.tsx

```diff
@@ -221,6 +253,16 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
         this._selectedItemViewMap[SelectedItemKeys.ReplicaSetKey] = (item) => this._getWorkoadPodsViewComponent(item.metadata, item.spec && item.spec.template, item.kind || "ReplicaSet", item.status.availableReplicas, item.status.replicas);
     }
 
+    private _populateObjectFinder() {
+        this._objectFinder[SelectedItemKeys.ReplicaSetKey] = (name) => (this._workloadsStore.getState().replicaSetList as V1ReplicaSetList).items.filter(r => r.metadata.name == name)[0];
+        this._objectFinder[SelectedItemKeys.StatefulSetKey] = (name) => (this._workloadsStore.getState().statefulSetList as V1StatefulSetList).items.filter(r => r.metadata.name == name)[0];
+        this._objectFinder[SelectedItemKeys.DaemonSetKey] = (name) => (this._workloadsStore.getState().daemonSetList as V1DaemonSetList).items.filter(r => r.metadata.name == name)[0];
+        this._objectFinder[SelectedItemKeys.ServiceItemKey] = (name) => {
+            let filteredServices: V1Service[] =  (this._servicesStore.getState().serviceList as V1ServiceList).items.filter(r => r.metadata.name == name);
+            return ServicesTable._getServiceItems(filteredServices)[0];
+        }
```

Comment:
should we have a common

this._workloadsStore.getState().replicaSetList as V1ReplicaSetList).items.filter(r => r.metadata.name == name)?

also, can we use ===,
also should not we check if the count is more than zero and return zero?